buffer/src/main/java/io/netty5/buffer/api/Buffer.java

```diff
@@ -399,6 +400,70 @@ default Buffer writeBytes(byte[] source, int srcPos, int length) {
         return this;
     }
 
+    /**
+     * Writes into this buffer from the source {@link ByteBuffer}.
+     * This updates the {@linkplain #writerOffset() write offset} of this buffer and also the position of
+     * the source {@link ByteBuffer}.
+     *
+     * @param source The {@link ByteBuffer} to read from.
+     * @return This buffer.
+     */
+    default Buffer writeBytes(ByteBuffer source) {
+        if (source.hasArray()) {
+            writeBytes(source.array(), source.arrayOffset() + source.position(), source.remaining());
+            source.position(source.limit());
+        } else {
+            int woff = writerOffset();
+            int length = source.remaining();
+            writerOffset(woff + source.remaining());
+            // Try to reduce bounce-checking by using long and int when possible.
+            boolean needReverse = source.order() != ByteOrder.BIG_ENDIAN;
```

Comment:
You can also temporarily change the order of the source to match the sink.

Reply:
I thought about this but I think this one should be faster and produce less GC.